### DISCUSSION TITLE
CNPG WAL Retention and Clean Up

### DISCUSSION THREAD

**Discussion by hspringm** [cloudnative-pg/cloudnative-pg#4216] [Q&A]
postgres pod instance, i see /var/lib/postgressql/wal/pg_wal contains old  3 to 4 months logs . Exploring options in CNPG to remove the logs automatically by postgres. Do we have option to control/setup WAL log retention ?  to cleanup if more than 15 days ? How to check current CNPG retention policy for WAL logs ?  archive_cleanup_command parameter is controlled by operator.  Please advise.

**Comment by gbartolini** [comment]
Please never touch files inside that folder. If they are there, it is for a reason. It normally depends on WAL size settings for PostgreSQL. Can you please share the content of the `/var/lib/postgresql/wal/pg_wal/archive_status` folder?

**Reply by hspringm** [reply]
Thank you, Below are the details.

/wal/pg_wal/archive_status$ ls -altr | head -10
-rw-rw---- 1 postgres tape      0 Feb  5 16:03 00000002.history.done
-rw-rw---- 1 postgres tape      0 Mar  4 16:12 00000003.history.done
-rw-rw---- 1 postgres tape      0 Mar 15 11:39 00000004.history.ready
-rw-rw---- 1 postgres tape      0 Mar 22 19:13 0000000400000020000000F1.000028.backup.ready
-rw-rw---- 1 postgres tape      0 Mar 22 19:14 0000000400000020000000F3.0000A0.backup.ready
-rw-rw---- 1 postgres tape      0 Mar 22 19:17 0000000400000020000000F5.000348.backup.ready
-rw-rw---- 1 postgres tape      0 Mar 28 19:54 0000000400000027000000D0.ready
-rw-rw---- 1 postgres tape      0 Mar 28 19:57 0000000400000027000000D1.ready
-rw-rw---- 1 postgres tape      0 Mar 28 19:59 0000000400000027000000D2.ready
postgres@cnpg-dev1:~/wal/pg_wal/archive_status$ ls -altr | tail -5
-rw------- 1 postgres tape      0 Apr  5 19:00 0000000600000041000052.ready
-rw------- 1 postgres tape      0 Apr  5 19:05 0000000600000041000053.ready
-rw------- 1 postgres tape      0 Apr  5 19:10 0000000600000041000054.ready

postgres@cnpg-dev1:~/wal/pg_wal/archive_status$ ls -altr | wc -l
2441

/wal/pg_wal$ du -csh .
39G    

Barman full backup is failing with 
:"barman-cloud-check-wal-archive","msg":"2024-04-05 13:45:01,359 [4013397] ERROR: WAL archive check failed for server cnpg-dev1: Expected empty archive"
"WARNING","sql_state_code":"01000","message":"archiving write-ahead log file \"00000004.history\" failed too many times, will try again later","backend_type":"archiver","query_id":"0"}}

Deployed CNPG instances with default parameter settings (separate storage for data & wal only)

**Reply by nleconte-csgroup** [reply]
Hello, I'm observing the same "issue" on one of my cluster, the wal folder is getting really big. Today the cluster could not start because the Kubernetes volume was full.

**Reply by ardentperf** [reply]
> Barman full backup is failing with
:"barman-cloud-check-wal-archive","msg":"2024-04-05 13:45:01,359 [4013397] ERROR: WAL archive check failed for server cnpg-dev1: Expected empty archive"

i think this means your backups arent working.

when a new cluster is created, i think the backups need an empty location for backups (for correct operation of barman around catalogs, restores, etc) and this message might indicate its pointing to a location thats not empty

if backups are fixed, then the write ahead logs should be automatically cleaned up
